### DISCUSSION TITLE
[feat suggestion]: partial/nullish/nullable object with default values

### DISCUSSION THREAD

**Discussion by jakecodev** [open-circle/valibot#767] [Ideas]
Hi, I found `partial` saves a lot of code using `optional` in object. It is great! But only `partial` works for object but could not set default values for object' keys. 

If I want a schema with default values '', it would be:

```ts
v.object({
  age: v.optional(v.string(), ''),
  role: v.optional(v.string(), 'visitor'),
  mobile: v.optional(v.string(), ''),
  // less readable when validation get complex
})
```

1. It would be cleaner if we can set default values at one time using v.partial
```ts
v.partial(
  v.object({
    age: v.string(),
    role: v.string(),
    mobile: v.string(),
  }),
  '',
)
// { age: '', role:'', mobile:'' ...}

```

 or with some exceptions
```ts
v.partial(
  v.object({
    age: v.number(),
    role: v.string(),
    mobile: v.string(),
    avatar: v.string(),
  }),
  '',
  {
    role: 'visitor',
    avatar: 'xxxxxxx.jpg',
  },
)
// { age: '', role:'visitor', mobile:'xxxxxxx.jpg' ...}
```

2. It would be great to extend v.nullable/nullish with the same functionality for objects.

This is just a suggestion. I believe this would be a feat that other users would love. Thank you again for this great library!

**Comment by fabian-hiller** [comment]
Your idea is great, but there are a few problems. If your object contains entries with different data types, it is not possible to define the default value with a single value, and your "exceptions" solution makes the API very complicated. Also, `partial` allows you to specify which keys should be optional via the second argument. Therefore, we can't define the default value with the second argument.
